src/Widgets/TerminalWidget.vala

```diff
@@ -525,33 +525,44 @@ namespace Terminal {
 
         // Check pasted and dropped text before feeding to child;
         private void validated_feed (string? text) {
-            if (text != null && text.validate ()) {
-                unowned var toplevel = (MainWindow) get_toplevel ();
-                if (!toplevel.unsafe_ignored &&
-                    Application.settings.get_boolean ("unsafe-paste-alert")) {
-
-                    string? warn_text = null;
-                    if ("\n" in text) {
-                        warn_text = _("The pasted text may contain multiple commands");
-                    } else if ("sudo" in text || "doas" in text) {
-                        warn_text = _("The pasted text may be trying to gain administrative access");
-                    }
+            // Do nothing when text is invalid
+            if (text == null || !text.validate ()) {
+                return;
+            }
 
-                    if (warn_text != null) {
-                        var dialog = new UnsafePasteDialog (toplevel, warn_text, text.strip ());
-                        dialog.response.connect ((res) => {
-                            dialog.destroy ();
-                            if (res == Gtk.ResponseType.ACCEPT) {
-                                feed_child (text.data);
-                            }
-                        });
+            unowned var toplevel = (MainWindow) get_toplevel ();
 
-                        dialog.present ();
-                    }
-                } else {
+            // No user interaction because of user's preference or previous selection
+            if (toplevel.unsafe_ignored ||
+                !Application.settings.get_boolean ("unsafe-paste-alert")) {
+
+                feed_child (text.data);
+                return;
+            }
+
+            string? warn_text = null;
+            if ("\n" in text) {
+                warn_text = _("The pasted text may contain multiple commands");
+            } else if ("sudo" in text || "doas" in text) {
+                warn_text = _("The pasted text may be trying to gain administrative access");
+            }
+
+            // No user interaction for safe commands
+            if (warn_text != null) {
```

Comment:
Should be `warn_text == null` ?

Reply:
Ah silly mistake…… :dizzy_face: I'll fix after #802 and #803 is get merged that should conflict with this PR. Thank you for pointing out this.